app/lib/preload.js

```diff
@@ -9,29 +9,82 @@
  */
 
 const {
-  remote,
+  contextBridge,
   ipcRenderer
 } = require('electron');
 
-const {
-  process,
-  app
-} = remote;
+const allowedEvents = [
```

Comment:
I wonder whether this will cause breaking changes for some plugins if we limit the allowed events.

Probably not, because we never actively promote using `backend` in your plugin directly.

Reply:
I'd say that calling the backend directly is not part of the public API. So I wouldn't care about the plugins which 
(ab)use the internals.

Reply:
Also, without an allowed list for the events, we open up a security loophole: https://blog.doyensec.com/2019/04/03/subverting-electron-apps-via-insecure-preload.html

Reply:
Valid point. Is it possible right now to setup a working frontend <-> backend interaction outside our existing communication channels? I believe not. I mean you could abuse the `menu` extension point for that. That one already sounds fishy for me though. 

My question is: Do we support backend <-> frontend plug-ins at all at the moment? If so, how do they look like? If not, fine to lock down this thing at the moment.

Reply:
The cloud connect plugin is a frontend & backend plugin of sorts. It saves templates to the config. Can't think of any other examples.

Reply:
https://github.com/bpmn-io/dmn-testing-plugin does it via fetch and `menu` abuse.

The only way to send a custom message to the backend is via [a private API created for Zeebe Plugin](https://github.com/camunda/camunda-modeler/blob/develop/client/src/plugins/zeebe-plugin/deployment-plugin/DeploymentPlugin.js#L71). Still, this is private API which is not marketed anywhere.

Reply:
So I can safely say that we don't support such communication.

Reply:
I don't know about such an (external) plugin, at least we don't have built one ourselves.

_Potentially_ it could be possible because we [expose the backend ](https://github.com/camunda/camunda-modeler/blob/develop/client/src/app/plugins/PluginsRoot.js#L93) to plugins. Then it _could_ be possible to use those events in the `menu` extension, as you said.

However, I think it's good if we pull that possibility down - if it exists at all.